sympy/functions/special/hyper.py

```diff
@@ -1043,3 +1045,42 @@ def _expr_big(cls, a, z, n):
     @classmethod
     def _expr_big_minus(cls, a, z, n):
         return -1/sqrt(1 + 1/z)*sinh(2*a*asinh(sqrt(z)) + 2*a*pi*I*n)
+
+class appellf1(Function):
+    '''
+    This is the Appell hypergeometric function of two variables as:
+    .. math ::
+        F_1(a,b_1,b_2,c,x,y) = \sum_{m=0}^{\infty} \sum_{n=0}^{\infty}
+        \frac{(a)_{m+n} (b_1)_m (b_2)_n}{(c)_{m+n}}
+        \frac{x^m y^n}{m! n!}.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Appell_series
+    .. [2] http://functions.wolfram.com/HypergeometricFunctions/AppellF1/
+
+    '''
+    @classmethod
+    def eval(cls, a, b1, b2, c, x, y):
+        if all(i.is_number for i in (a, b1, b2, c, x, y)):
+            return S(apl(a, b1, b2, c, x, y))
```

Comment:
I don't think this is the proper way to get automatic numeric evaluation.

Reply:
can't we use mpmath's `appellf1` to perform numeric evaluation ?

Reply:
Well, mpmath is for `.evalf`, not for `.eval`.

Reply:
It should already work automatically.

```
>>> class appellf1(Function):
...     nargs = 6
>>> appellf1(1., 2., 3., 4., 5., 6.)
-0.151473567018263
```

That's because Function subclasses automatically work with evalf if they use the same name as their mpmath counterpart, and `_should_evalf` (which is the proper way to do this) defaults to evalfing on float inputs.